### ISSUE TITLE
Create running Triggers in cloud build for dev and main

### ISSUE THREAD

**Issue by PaulWGeorge** [Healthy-Plate-Project/Healthy-Eating-App#81]

**Comment by PaulWGeorge** [comment]
Current issue for trigger to solve: 

![Image](https://user-images.githubusercontent.com/88292917/189008688-6fcca14d-de19-4894-8c3f-1bfd9d9b82b2.png)

**Comment by PaulWGeorge** [comment]
Remaking a new config file for deploy: https://cloud.google.com/build/docs/deploying-builds/deploy-appengine

**Comment by PaulWGeorge** [comment]
Something is up with logging in cloud build:
![Image](https://user-images.githubusercontent.com/88292917/189550623-2d744f85-a006-4a6e-a6c3-7af3786373a4.png)

**Comment by PaulWGeorge** [comment]
Will try to add service account api to be enabled:


![Image](https://user-images.githubusercontent.com/88292917/189550645-062c7e20-60f3-4b3b-964f-674b7427186e.png)

**Comment by PaulWGeorge** [comment]
I do not need to make new credentials:

![Image](https://user-images.githubusercontent.com/88292917/189550677-25c68120-8a3f-4a61-8e23-122b6751d4e2.png)

**Comment by PaulWGeorge** [comment]
Disabled the gcp api since it did not fix anything.

**Comment by PaulWGeorge** [comment]
Seems like service account for cloud build should be able to create the logs: 

![Image](https://user-images.githubusercontent.com/88292917/189550815-bc5755ce-e755-40d9-aa14-5a2c8008e4b2.png)

**Comment by PaulWGeorge** [comment]
Documentation on deploying to app engine using cloud build: https://cloud.google.com/build/docs/deploying-builds/deploy-appengine

I added two roles to the google cloud build service account:

![Image](https://user-images.githubusercontent.com/88292917/189576695-e9d09fc4-b309-439c-b217-58cf5ed7afef.png)

**Comment by PaulWGeorge** [comment]
Current service account permissions: 

![Image](https://user-images.githubusercontent.com/88292917/189576896-9021ef24-ff2c-4d89-af0c-c809b8bfdb52.png)

**Comment by PaulWGeorge** [comment]
Attempting to run trigger gives me this: 

![Image](https://user-images.githubusercontent.com/88292917/189578537-2ba6d0b1-3f85-46d5-ac1c-934e19bb4594.png)

**Comment by PaulWGeorge** [comment]
Enabling IAM API: 

![Image](https://user-images.githubusercontent.com/88292917/189578741-11d4d0d6-6dfd-4e4a-9559-ba76ce493e8b.png)

**Comment by PaulWGeorge** [comment]
Trigger finally runs, but another API needs to be enabled:

![Image](https://user-images.githubusercontent.com/88292917/189579223-969d1a8a-f806-4032-9aa8-f4cd0ea12152.png)

**Comment by PaulWGeorge** [comment]
This is the API I am enabling: 

![Image](https://user-images.githubusercontent.com/88292917/189579286-6d246d6f-6f82-4875-9cf5-cac969eb48a5.png)

**Comment by PaulWGeorge** [comment]
Github check  passed:

![Image](https://user-images.githubusercontent.com/88292917/189580151-1de11661-507c-4d30-a779-c1da506f7d23.png)

Cloud Build executed with success: 

![Image](https://user-images.githubusercontent.com/88292917/189580221-0c1f6f28-c553-4e23-9062-e4bb73caf007.png)

App Engine successfully ran the repo contents: 

![Image](https://user-images.githubusercontent.com/88292917/189580301-0283c77c-317e-42e7-bf27-1206825dd384.png)

Traffic has moved over to newest test version: 

![Image](https://user-images.githubusercontent.com/88292917/189580371-18b56181-f9ae-4d30-a41f-5e1e51817750.png)

Unfortunately the webpage is not pulling anything: 

![Image](https://user-images.githubusercontent.com/88292917/189580421-13be7028-e48c-49db-82d3-52b40a555104.png)

**Comment by PaulWGeorge** [comment]
Still getting the same errors, and I have reached a blocker: 
This is the link to the version page: https://console.cloud.google.com/appengine/versions?serviceId=default&versionId=20220912t052449&authuser=3&project=healthy-eating-project-359101

This is the build page: https://console.cloud.google.com/cloud-build/builds;region=us-central1?project=healthy-eating-project-359101

This is for logs explorer: https://console.cloud.google.com/logs/query;cursorTimestamp=2022-09-14T03:00:00.000184Z?serviceId=default&versionId=20220912t052449&authuser=3&project=healthy-eating-project-359101

**Comment by PaulWGeorge** [comment]
@johndavis92790  Can you take a look at this "Cannot Get /" error and let me know if you have any ideas? I just need another pair of eyes on this. Thanks!

**Comment by johndavis92790** [comment]
@PaulWGeorge What is the exact url that you are trying to access when getting the "Cannot Get /" error?

**Comment by PaulWGeorge** [comment]
Should be the exact one that the client is trying to access locally when it runs the app

**Comment by johndavis92790** [comment]
Sorry I am not at my computer so I can't really see or do anything other than what's on github right now. I'm pretty sure I know what the problem is but I can't test my theory right now. Just paste the entire url from the address bar in your browser when you get the error into this chat, then I can tell you if my theory is correct or if it's something else.

**Comment by PaulWGeorge** [comment]
This is the url for the address: https://healthy-eating-project-359101.uc.r.appspot.com/

**Comment by johndavis92790** [comment]
Ok try removing the "/" at the end of the url and press enter.

**Comment by PaulWGeorge** [comment]
Oh weird it is not in the address but it is when I paste it

**Comment by johndavis92790** [comment]
Ya thats a browser thing, I think our app routing code is not written correctly. I will check it out and push some changes to the develop branch. Give me a few mins

**Comment by PaulWGeorge** [comment]
All good. Has it been working on local builds?

**Comment by johndavis92790** [comment]
Yes these things run differently on local versus from the server. I just pushed new changes to the develop branch, let me know if that works.

**Comment by PaulWGeorge** [comment]
I will run that on my branch

**Comment by johndavis92790** [comment]
Wait I thought these triggers are connected to the develop branch?

**Comment by johndavis92790** [comment]
You are probably going to run into a bunch of merge conflicts if you try and pull from develop into your own branch

**Comment by PaulWGeorge** [comment]
I already solved the merge conflicts a while ago and resolved them. No worries. I just have one file I needed

**Comment by PaulWGeorge** [comment]
The trigger is connected to develop but it does not have my config file I made. I will push that to develop once I get this working

**Comment by johndavis92790** [comment]
Ok cool, I only changed two lines of code in the `App.tsx` file

**Comment by PaulWGeorge** [comment]
Same error unfortunately

**Comment by johndavis92790** [comment]
Ok, hold on a sec

**Comment by PaulWGeorge** [comment]
On my local branch when I do "npm run develop"  I get this when testing local:


![Image](https://user-images.githubusercontent.com/88292917/190061665-26752961-b1e3-4fea-beb9-ff99561e6ef4.png)

**Comment by johndavis92790** [comment]
That is a different issue, the merge conflicts were probably not merged correctly. I just pushed new changes again to develop. This will be easier if the develop branch had that config file, can you push email me that file so I can push it directly to develop? Its ok if it doesn't work, no one is using that branch yet anyways

**Comment by johndavis92790** [comment]
johndavis92790@gmail.com

**Comment by PaulWGeorge** [comment]
I will just do a PR to develop with my file in a minute

**Comment by johndavis92790** [comment]
Well do a PR from your own branch into develop. Only push into your own branch, do not push into develop directly.

**Comment by PaulWGeorge** [comment]
I have pushed mine. The file is called cloudbuild.yaml

**Comment by johndavis92790** [comment]
I can't see anything until you create a PR from that push, just go to the "Pull Requests" tab in the main github page for the repo., you should see a banner at the top to create a PR, follow the steps in the wiki.

**Comment by johndavis92790** [comment]
https://github.com/Healthy-Plate-Project/Healthy-Eating-App/wiki/Basic-development-steps
Follow step 10 and on.

**Comment by PaulWGeorge** [comment]
So you do want me to do it for my branch to develop?

**Comment by johndavis92790** [comment]
You should only ever push into your own branch, it won't ever let you do it into develop anyways, it has protections turned on.

**Comment by PaulWGeorge** [comment]
Right. I cannot do a PR from my branch into my own branch though

**Comment by johndavis92790** [comment]
Sorry I see why my comment was confusing, the PR is from your branch into develop. You should only push into your own branch in git locally, the merging into develop is done in the github cloud, not git locally. So once you have created the PR, then I can see the changes and go from there. I have updated that comment haha sorry

**Comment by johndavis92790** [comment]
I've got to go to bed now, I will take a look tomorrow, thanks for working on this!

**Comment by PaulWGeorge** [comment]
Me as well. Thanks!

**Comment by PaulWGeorge** [comment]
New error coming from develop:


![Image](https://user-images.githubusercontent.com/88292917/190281023-4bc6742f-dc4b-4922-b155-7a9778961561.png)

@johndavis92790

**Comment by PaulWGeorge** [comment]
Error code 500 Server Error

**Comment by PaulWGeorge** [comment]
How to create specific deployment for build: https://cloud.google.com/appengine/docs/flexible/nodejs/testing-and-deploying-your-app

**Comment by PaulWGeorge** [comment]
![Image](https://user-images.githubusercontent.com/88292917/190288045-259c184f-1e7a-4709-998f-13d5f1afee0c.png)

The top line is not receiving traffic and from demo. This is great. We can still view it, but the public does not see it
